django/http/request.py

```diff
@@ -21,13 +22,11 @@
 from django.utils.functional import cached_property
 from django.utils.http import is_same_domain
 from django.utils.regex_helper import _lazy_re_compile
-from django.utils.version import PY38
 
 from .multipartparser import parse_header
 
-if PY38:
-    from urllib.parse import parse_qsl
-else:
+# TODO: Remove when Python 3.8 is the minimum required version for Django.
+if 'max_num_fields' not in inspect.signature(parse_qsl).parameters:
     from django.utils.http import parse_qsl
 
```

Comment:
I guess I'm happy with the `inspect` usage. (Is it too cute?) I'd probably add a comment explaining the why. 
I'll give it a run with an old version to double check... 

I'd rather keep the compatibility import as a single `if...else...` block. Having the main import above and then replacing here isn't as obvious as it could be, for me.

Reply:
It cannot be an `if ... else ...` block as `inspect.signature()` needs to look at the signature of `parse_qsl()` from `urllib.parse` which means it needs to be imported first. Hence this override if required approach. I also didn't want to move the import from the standard library just above the `if` otherwise we'd just be hacking around with exclusion comments for `isort`.

Reply:
Comment added.

Reply:
Grrr... I think I probably preferred the `if PY38` version then. (Let me confer with Mariusz.)